src/ape/cli/commands.py

```diff
@@ -1,26 +1,43 @@
 import inspect
 import warnings
-from typing import Any, List
+from typing import Any, List, Optional
 
 import click
 from click import Context
 
 from ape import networks
-from ape.api import ProviderAPI
+from ape.api import ProviderAPI, ProviderContextManager
 from ape.cli.choices import NetworkChoice
 from ape.exceptions import NetworkError
 
 
-def check_parents_for_interactive(ctx: Context) -> bool:
-    interactive: bool = ctx.params.get("interactive", False)
```

Comment:
bug 1: was defaulting to False here when it shouldnt be because it hasnt checked the parents yet.. you can see below we check for parents only if None here (unspecified)